.github/workflows/reports.yml

```diff
@@ -316,17 +310,10 @@ jobs:
           export PATH="$PATH:$(dirname $nargo_binary)"
           nargo -V
 
-      - name: Download compilation report script
-        uses: actions/download-artifact@v4
+      - uses: actions/checkout@v4
         with:
-          name: script_for_compilation_report
-          path: ./test-repo/${{ matrix.project.path }}
-
-      # - name: Download compilation report
-      #   uses: actions/download-artifact@v4
-      #   with:
-      #     name: in_progress_compilation_report
-      #     path: ./test-repo/${{ matrix.project.path }}
+          sparse-checkout: |
+            ./test_programs/compilation_report.sh
```

Comment:
We're downloading this script to a different directory now, I'm not sure if we can do the checkout into `./test_repo` or if we need to move the file after the fact however.

We may need to update this to not use cone mode: https://github.com/actions/checkout?tab=readme-ov-file#fetch-only-a-single-file

Reply:
I imagine that it's placed at `test_repo/test_programs/compilation_report.sh` so that if you were to add more files into the sparse checkout then it won't move any of the old files.

Reply:
Yeah I just noticed that I was not specifying cone mode as false.

Reply:
> I'm not sure if we can do the checkout into `./test_repo` or if we need to move the file after the fact however.

I guess we shall see